### ISSUE TITLE
Un-tracking of completed quests not working!

### ISSUE THREAD

**Issue by Kiefy** [tyra314/SmartQuestTracker#1]
I have only this mod loaded. absolutely nothing else.

"Completed Quests" is selected in options, yet they are not removed when I reloadui, change zones etc.
![Screenshot](https://image.prntscr.com/image/F9vQvkjRRVaJXqvcGTJNEg.png)
Clicking Force update does nothing. If I untrack all quests, then change zone or force update, it adds the completed quests back into the tracker.

**Comment by Kiefy** [comment]
Ok. I found out why. There were 2 reasons it wasn't working.

1) There is a typo in your code. you first created the var `isComplete`, but then attempt to read `isCompleted`.

2) You also need to add `untrackAllQuests()` to the beginning of your `run_update()` function, else the watchframe doesn't get updated.  Which will fix the issue the other guy in comments was having.

**Comment by tyra314** [comment]
> There is a typo in the code. you first create the var `isComplete`, but then attempt to read `isCompleted`.
Uh, I have a deja-vu :(

> I also had to add untrackAllQuests() to the beginning of the run_update() function, else the watchframe never gets updated. Which should also fix the issue the other guy was having. I replied to his post to let him and others know.
No, that does WAY to much. It also deletes the information about autotracked and thus not autotracked quests.

> I also made it do a run_update() when completing a quest, so it would be removed immediately, instead of when changing zone.
That was actually on purpose, so you have a chance to see which quest was just completed.

**Comment by Kiefy** [comment]
Yeah. I'm sure you can do a better job of fixing it. but it at least my dodgy hacks make it work the way I want it to for now.

I also use MonkeyQuest to see overall status. I just needed the blizz tracker active to get quest blobs on the minimap, so I didn't need to see completed at all in there. but my heavy-handed attempt to make it update immediately doesn't work all the time anyway. maybe being called too soon.

Love the addon though. it answered all my quest tracking prayers and I look forward to your cleaner fix. thank you! :)

**Comment by tyra314** [comment]
An update should be available as soon as curse gets it processed.... Thanks for the bug hunt.

**Comment by Kiefy** [comment]
There is still the issue of the tracker not updating unless you force it.

my method may have been overhanded, but something needs to update it.

**Comment by Kiefy** [comment]
I don't get it. I just did the Frostlord Ahune boss on my Warlock and it correctly cleared the Redridge quests I had.

The only difference I can tell is that Frostlord Ahune doesn't put any Dungeon Objectives in the Watchlist.

**Comment by tyra314** [comment]
Whenever this happens next, please use the "print all auto tracked quests" button and post the output. The thing is, I've seen this myself and every single time it was a problem that the auto tracking was confused and forgot, which quests have been tracked by the addon and which were tracked by the user. However, I never found a culprit for that behavior, because it usually works good enough.

**Comment by tyra314** [comment]
I've pushed a new alpha version which reworks quite a bit and should help performance wise.
https://wow.curseforge.com/projects/smartquesttracker/files/2578377

However, I've no idea on the impact on this issue.
